src/Aspire.Dashboard/Components/_Imports.razor

```diff
@@ -10,3 +11,6 @@
 @using Aspire.Dashboard.Components
 @using Aspire.Dashboard.Components.Controls
 @using Microsoft.Extensions.Localization
+
+@* Require authorization for all pages *@
+@attribute [Authorize]
```

Comment:
Do we show the user information anywhere?

Reply:
Not yet. I would like to, along with a sign out button.

I didn't find a way to perform OIDC sign out from Blazor server though. Everything I saw doc'd revolved around using HttpContext in action handler methods, which doesn't work in Blazor server, afaict, throwing errors about the headers being read only because the response has started.

Reply:
AFAIK you can use the AuthenticationStateProvider

Reply:
BTW why is this always on? We don't have auth when running locally right?

Reply:
Should we make this more explicit by creating an authorization policy and referring to it here?

Reply:
> why is this always on?

My observation was that `[Authorize]` only does anything when you actually have authorization middleware configured. The dashboard only does that when you opt-in to the `OpenIdConnect` auth mode.

Reply:
Isn't it always turned on here? https://github.com/dotnet/aspire/pull/3033/files#diff-ccedb77216895077db571ce0a3c547948ca85798f4ce0523e4eb936216f27f15L196

Reply:
> I can look into that. What benefit would that have?

It's a bit confusing now that there's so many authentication schemes being configured, and different schemes between the OTLP endpoint and the rest of the dashboard UI. Having the UI explicitly configured to require authn/z based on the scheme that was configured for dashboard auth will help make it clearer IMO.

Reply:
Specify a policy name on the attribute. Makes it easier to understand if each part of the dashboard has a named policy for auth.

OTLP services is already doing this.

Reply:
https://github.com/dotnet/aspire/blob/0783a609d369634c1a0aa5d399b78c79a52caafe/src/Aspire.Dashboard/Otlp/Grpc/OtlpLogsService.cs#L14

Reply:
A policy's been added for the web app's HTTP traffic. Now, when OpenIdConnect is selected, requests are challenged correctly. OTLP routes to its authentication and policy.

Today I learned what authentication schemes and authorization policies are, then produced the update here. The approach looks good (for all I know) but would appreciate expert reviews from y'all.

Reply:
@JamesNK I removed the default authentication scheme and made the OTLP policy reference that scheme explicitly.

Reply:
Have captured the ideas here in a TODO list in the PR description.